x-pack/plugins/reporting/server/config/schema.ts

```diff
@@ -138,6 +145,7 @@ export const ConfigSchema = schema.object({
   roles: RolesSchema,
   poll: PollSchema,
   export_types: ExportTypeSchema,
+  enableStatefulSettings: SettingsSchema,
```

Comment:
The full path for the setting is `xpack.reporting.enableStatefulSettings.enabled: true|false`. How about shortening it with something like:
* `xpack.reporting.enableStatefulSettings: true|false`
* `xpack.reporting.statefulSettings.enabled: true|false`

I think my preference is for the 2nd one.